### DISCUSSION TITLE
Database changes when unlocked vs locked

### DISCUSSION THREAD

**Discussion by beyond9thousand** [Kunzisoft/KeePassDX#1527] [Q&A] [answered]
So I use syncthing to keep the database synchronised between my PC and my phone. I was wondering how keepassdx behaves with that? If the database remains unlocked and it's overwritten by new changes (assuming the app is closed) does it automatically handle the change? If not, what is the ideal way to go about things? 

Re-unlocking the database each time I need to enter my credentials anywhere on the phone feels a tad bit slow.

**Comment by J-Jamet** [comment, marked as answer]
KeePassDX uses a very simple workflow:
 - The application doesn't keep any cache when a database is closed, so as soon as a database file URI link is opened, it just reads the last data that is sent by the file provider of the used file manager. Syncthing synchronizes the files but I don't think it provides them to KeePassDX, it is therefore common to go through the system file provider (content://com.android.externalstorage.documents/).
 - KeePassDX detects a file change only by the URI metadata, which means that the file provider has to modify this data (this is the case with the system one). Once detected, a message is displayed indicating that the file data has been changed, you can either merge it, ignore the local data and reload the file data, or ignore the file data. https://github.com/Kunzisoft/KeePassDX/wiki/File-Manager-and-Sync

So to answer your question, opening a (previously closed) file in KeePassDX guarantees that it is the latest version provided by the file manager. But if it is currently open, it is up to you to choose the merge method to apply.

**Comment by beyond9thousand** [comment]
Thank you for the explanation :)
